src/bin/utils/sync.ts

```diff
@@ -0,0 +1,179 @@
+import { confirm } from '@inquirer/prompts';
+
+import { generateFieldId } from '@/src/bin/utils/schema';
+import type { Schema } from '@/src/types/schema';
+
+/**
+ * Fetch the available spaces for the current session.
+ *
+ * @param sessionToken The session token to use for the request.
+ *
+ * @returns The available spaces for the current session.
+ */
+export const getSpaces = async (
+  sessionToken: string,
+): Promise<{ id: string; handle: string; name: string }[]> => {
+  let res;
+  try {
+    res = await fetch('https://ronin.supply/-/ronin/spaces', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${sessionToken}`,
+      },
+    });
+  } catch (err) {
+    throw new Error('Failed to fetch available spaces: ${err.message}');
+  }
+
+  const text = await res.text();
+
+  let json;
+  try {
+    json = JSON.parse(text);
+  } catch (err) {
+    throw new Error(`Failed to parse response from server: ${text}`);
+  }
+
+  return json.spaces;
+};
+
+/**
+ * Fetch the existing schemas for the configured space.
+ *
+ * @param token Either an App Token or a Session Token.
+ * @param space The ID of the space to fetch schemas for.
+ *
+ * @returns The schemas of the provided space
+ */
+export const getSchemas = async (token: string, space: string): Promise<Schema[]> => {
+  let res;
+  try {
+    res = await fetch(`https://ronin.supply/-/ronin/schemas?data-selector=${space}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
+      },
+    });
+  } catch (err) {
+    throw new Error('Failed to fetch available spaces: ${err.message}');
+  }
+
+  const text = await res.text();
+
+  let json;
+  try {
+    json = JSON.parse(text);
+  } catch (err) {
+    throw new Error(`Failed to parse response from server: ${text}`);
+  }
+
+  return json.schemas;
+};
+
+/**
+ * Replace the ID's of fields in the local schema with the ID's
+ * of the existing fields in the remote schema.
+ *
+ * @param local Local schema definitions.
+ * @param remote Existing remote schema definitions.
+ *
+ * @returns Updated local schema definitions.
+ */
```

Comment:
```suggestion
/**
 * Replace the IDs of fields in the local schema with the IDs
 * of the existing fields in the remote schema.
 *
 * @param local Local schema definitions.
 * @param remote Existing remote schema definitions.
 *
 * @returns Updated local schema definitions.
 */
```

"ID's" means something is owned by the IDs. For example: "The ID's car", meaning a car owned by an ID. If you just want to refer to them, you need to use "IDs", just like you use "cars" and not "car's".

Reply:
Thanks for the explanation! I'm aware of what it should mean. Just eslint was complaining so I used that.

Reply:
Sure, but that doesn't make it any more correct. Sometimes ESLint doesn't work as expected, in which case we need to ignore certain words (like you did).